plugins/graphql-kotlin-plugin-core/src/main/kotlin/com/expediagroup/graphql/plugin/generator/types/generateTypeName.kt

```diff
@@ -148,21 +151,28 @@ private fun isCachedTypeApplicable(context: GraphQLClientGeneratorContext, graph
 
 private fun verifySelectionSet(context: GraphQLClientGeneratorContext, graphQLTypeName: String, selectionSet: SelectionSet): Boolean {
     val selectedFields = calculateSelectedFields(context, graphQLTypeName, selectionSet)
-    val typeSpec = context.typeSpecs[graphQLTypeName]
-    val properties = typeSpec?.propertySpecs?.map { it.name }?.toSet() ?: emptySet()
-    return selectedFields == properties ||
-        (selectedFields.minus(properties).size == 1 && selectedFields.contains("__typename") && context.objectsWithTypeNameSelection.contains(graphQLTypeName))
+    val properties = calculateGeneratedTypeProperties(context, graphQLTypeName)
+    if (context.objectsWithTypeNameSelection.contains(graphQLTypeName)) {
+        properties.add("__typename")
+    }
+    return selectedFields == properties
 }
 
 private fun calculateSelectedFields(
     context: GraphQLClientGeneratorContext,
     targetType: String,
-    selectionSet: SelectionSet
+    selectionSet: SelectionSet,
+    path: String = ""
 ): Set<String> {
     val result = mutableSetOf<String>()
     selectionSet.selections.forEach { selection ->
         when (selection) {
-            is Field -> result.add(selection.name)
+            is Field -> {
+                result.add(path + selection.name)
+                if (selection.selectionSet != null) {
+                    result.addAll(calculateSelectedFields(context, targetType, selection.selectionSet, "$path${selection.name}."))
```

Comment:
Is this supposed to be `"$path.${selection.name}" `?

Reply:
path is initialize as empty string and then we pass it as `path + selection + .` so current logic is correct (otherwise we end up with `.` prefix)

Reply:
What is the end result here because `pathfoo.` doesn't seem like a correct string

Reply:
it is a prefix for recursive selection, `path` is a prefix
```
[0] prefix = "", field = "foo" -> "foo"
[1] prefix = "foo.", field = "bar" -> "foo.bar"
[2] prefix = "foo.bar.", field = "baz" -> "foo.bar.baz"
```